lib/metadataTypes/Asset.js

```diff
@@ -476,26 +476,106 @@ class Asset extends MetadataType {
+     *
+     * @param {string} code code from extracted code
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
+     * @returns {string} code with markets applied
+     */
+    static _applyMarketValues(code, templateVariables) {
+        // replace template variables with their values
+        return Mustache.render(code, templateVariables);
+    }
+    /**
+     * helper for buildTemplateForExtracts
+     * searches extracted extracted files for template variables and applies the market variable names
+     *
+     * @param {string} code code from extracted code
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
+     * @returns {string} code with markets applied
+     */
+    static _applyTemplateVariables(code, templateVariables) {
+        // replace template variables with their values
+        return Util.replaceByObject(code, templateVariables);
+    }
     /**
      * helper for buildDefinition
      * handles extracted code if any are found for complex types
      *
      * @param {string} templateDir Directory where metadata templates are stored
      * @param {string} targetDir Directory where built definitions will be saved
      * @param {TYPE.AssetItem} metadata main JSON file that was read from file system
-     * @param {TYPE.TemplateMap} variables variables to be replaced in the metadata
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
      * @param {string} templateName name of the template to be built
      * @returns {Promise.<void>} Promise
      */
-    static async buildDefinitionForExtracts(
+    static buildDefinitionForExtracts(
         templateDir,
         targetDir,
         metadata,
-        variables,
+        templateVariables,
         templateName
     ) {
-        // clone metadata to ensure the main file is not modified by what we do in here
+        return this._buildXForExtracts(
+            templateDir,
+            targetDir,
+            metadata,
+            templateVariables,
+            templateName,
+            'definition'
+        );
+    }
+    /**
+     * helper for buildTemplate
+     * handles extracted code if any are found for complex types
+     *
+     * @param {string} templateDir Directory where metadata templates are stored
+     * @param {string|string[]} targetDir (List of) Directory where built definitions will be saved
+     * @param {TYPE.AssetItem} metadata main JSON file that was read from file system
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
+     * @param {string} templateName name of the template to be built
+     * @returns {Promise.<void>} void
+     */
+    static buildTemplateForExtracts(
+        templateDir,
+        targetDir,
+        metadata,
+        templateVariables,
+        templateName
+    ) {
+        return this._buildXForExtracts(
```

Comment:
can we have a chat on this logic so i get it?

Reply:
Before the talk: we already had a buildDefinitionForExtract. That included the logic which files to look for. The only diff to what we needed for buildTEMPLATE was running mustache vs. replacing dev value with market variable names